tensorrt_llm/_torch/pyexecutor/py_executor.py

```diff
@@ -806,26 +803,21 @@ def _executor_loop(self):
         with self._profiler() as profile_step:
             iter_start_time = time.time()
             iter_stats = None
-            while not got_finish_signal or len(self.active_requests) > 0:
+            while True:
                 profile_step()
                 if self.enable_iter_perf_stats:
                     iter_start_time = time.time()
-                new_requests = self._fetch_new_requests()
-                got_finish_signal = self._merge_requests(
-                    new_requests) or got_finish_signal
-                if got_finish_signal and len(self.active_requests) == 0:
-                    break
+                new_requests_num = self._fetch_new_requests()
 
                 if self.kv_cache_transceiver:
                     self._check_disagg_gen_transfer_status()
 
                 if self.enable_iter_perf_stats:
                     iter_stats = self._get_init_iter_stats(
-                        len(new_requests),
+                        new_requests_num,
                         self.new_active_requests_queue_latency_ms)
 
-                if not got_finish_signal:
-                    num_dummy_request = self._get_num_dummy_request()
+                num_dummy_request = self._get_num_dummy_request()
                 if num_dummy_request > 0:
```

Comment:
Can this if check be moved to inside `_merge_dummy_request`? `_merge_dummy_request` should be able to handle `num_dummy_request=0`